packages/dosespot-react/src/useDoseSpotNotifications.ts

```diff
@@ -0,0 +1,75 @@
+import { useMedplum } from '@medplum/react-hooks';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { DOSESPOT_NOTIFICATION_COUNTS_BOT, DoseSpotNotificationCountsResponse } from './common';
+
+export interface DoseSpotNotificationsOptions {
+  readonly refreshIntervalMilliseconds?: number;
+  readonly onChange?: (count: number) => void;
+  readonly onError?: (err: unknown) => void;
+}
+
+const DEFAULT_REFRESH_INTERVAL_MILLISECONDS = 10000;
+
+export function useDoseSpotNotifications(options?: DoseSpotNotificationsOptions): number | undefined {
+  const medplum = useMedplum();
+  const { onChange, onError } = options ?? {};
+  const hasDoseSpot = medplum.getProjectMembership()?.identifier?.some((i) => i.system?.includes('dosespot'));
+  const refreshInterval = options?.refreshIntervalMilliseconds ?? DEFAULT_REFRESH_INTERVAL_MILLISECONDS;
+  const timerRef = useRef<NodeJS.Timeout | undefined>(undefined);
+  const [unreadCount, setUnreadCount] = useState<number | undefined>(undefined);
+
+  const stopTimer = useCallback(() => {
+    const timerId = timerRef.current;
+    if (timerId) {
+      clearInterval(timerId);
+    }
+  }, []);
+
+  const updateCount = useCallback(async () => {
+    try {
+      const result = (await medplum.executeBot(
+        DOSESPOT_NOTIFICATION_COUNTS_BOT,
+        {}
+      )) as DoseSpotNotificationCountsResponse;
+
+      let newCount = 0;
+      if (result.PendingPrescriptionsCount !== undefined) {
+        newCount += result.PendingPrescriptionsCount;
+      }
+      if (result.PendingRxChangeCount !== undefined) {
+        newCount += result.PendingRxChangeCount;
+      }
+      if (result.RefillRequestsCount !== undefined) {
+        newCount += result.RefillRequestsCount;
+      }
+      if (result.TransactionErrorsCount !== undefined) {
+        newCount += result.TransactionErrorsCount;
+      }
+      if (newCount !== unreadCount) {
+        setUnreadCount(newCount);
+        onChange?.(newCount);
```

Comment:
Interesting, why have both `unreadCount` and also accept a function for `onChange`?

Not super icked by it, just seems like these both do the same thing, ie.:

```tsx
  // This
  const unreadCount = useDoseSpotNotifications();
  useEffect(() => {
     doSomething();
  }, [unreadCount]);
  
  // is the same thing as this
 useDoseSpotNotifications({ onChange: () => doSomething() });
```

Reply:
Yeah.... I debated that.   First one probably better?

Reply:
Yeah I could probably go either way... maybe it's actually fine to have both?

I do prefer the first one. But we already have `onError` either way, so for symmetry I could see `onChange` also existing...

But from the perspective of DX / simplicity I do like the idea of having one way to do everything...

I vote for 1 but if we keep both I wouldn't revolt

Reply:
Leaving both for now, happy to revisit if we don't like it.